src/imaCore.mjs

```diff
@@ -881,90 +694,65 @@ async function handleAllMessagesSigning( optsTransfer ) {
-                        );
+                    const ethersProviderNode = owaspUtils.getEthersProviderFromURL( strUrlHttp );
+                    const joMessageProxyNode = new owaspUtils.ethersMod.ethers.Contract(
+                        sc.joAbiIMA.message_proxy_chain_address,
+                        sc.joAbiIMA.message_proxy_chain_abi,
+                        ethersProviderNode );
                     const strEventName = "OutgoingMessage";
                     const node_r = await imaEventLogScan.safeGetPastEventsProgressive(
-                        optsTransfer.details, optsTransfer.strLogPrefixShort,
-                        ethersProviderNode, 10, joMessageProxyNode, strEventName,
+                        optsTransfer.details, optsTransfer.strLogPrefixShort, ethersProviderNode,
+                        10, joMessageProxyNode, strEventName,
                         joMessage.savedBlockNumberForOptimizations,
                         joMessage.savedBlockNumberForOptimizations,
                         joMessageProxyNode.filters[strEventName](
                             owaspUtils.ethersMod.ethers.utils.id( optsTransfer.chainNameDst ),
-                            owaspUtils.toBN( idxImaMessage )
-                        )
-                    );
+                            owaspUtils.toBN( idxImaMessage ) ), optsTransfer.optsChainPair );
                     const cntEvents = node_r.length;
-                    if( log.verboseGet() >= log.verboseReversed().trace ) {
-                        optsTransfer.details.write( optsTransfer.strLogPrefix + cc.debug( "Got " ) +
-                            cc.info( cntEvents ) + cc.debug( " event(s) (" ) +
-                            cc.info( strEventName ) + cc.debug( ") on node " ) +
-                            cc.info( joNode.name ) + cc.debug( " with data: " ) +
-                            cc.j( node_r ) + "\n" );
-                    }
+                    optsTransfer.details.trace( "{p}Got {} event(s) ({}) on node {} with data: {}",
+                        optsTransfer.strLogPrefix, cntEvents, strEventName, joNode.name, node_r );
                     for( let idxEvent = 0; idxEvent < cntEvents; ++ idxEvent ) {
                         const joEvent = node_r[idxEvent];
                         const eventValuesByName = {
-                            "dstChainHash": joEvent.args[0],
-                            "msgCounter": joEvent.args[1],
-                            "srcContract": joEvent.args[2],
-                            "dstContract": joEvent.args[3],
-                            "data": joEvent.args[4]
+                            "dstChainHash": imaEventLogScan.extractEventArg( joEvent.args[0] ),
+                            "srcContract": imaEventLogScan.extractEventArg( joEvent.args[2] ),
+                            "dstContract": imaEventLogScan.extractEventArg( joEvent.args[3] ),
+                            "data": imaEventLogScan.extractEventArg( joEvent.args[4] )
```

Comment:
Is `imaEventLogScan.extractEventArg` still relevant?

Reply:
Removed everywhere, done